chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -18,6 +22,31 @@
 #undef RenderViewContextMenu
 #define RenderViewContextMenu RenderViewContextMenu_Chromium
 
+namespace {
+
+GURL GetSelectionNavigationURL(Profile* profile, const base::string16& text) {
+  AutocompleteMatch match;
+  AutocompleteClassifier classifier(
+      std::make_unique<BraveAutocompleteController>(
+          std::make_unique<BraveAutocompleteProviderClient>(profile),
+          nullptr, AutocompleteClassifier::DefaultOmniboxProviders()),
+      std::make_unique<BraveAutocompleteSchemeClassifier>(profile));
+  classifier.Classify(text, false, false,
+                      metrics::OmniboxEventProto::INVALID_SPEC, &match, NULL);
+  classifier.Shutdown();
+  return match.destination_url;
+}
+
+}  // namespace
+
+#define BRAVE_APPEND_SEARCH_PROVIDER \
+  if (GetProfile()->IsOffTheRecord()) { \
+    selection_navigation_url_ = \
```

Comment:
I'm confused here, the selection_navigation_url_ is set from `AutocompleteClassifierFactory::GetForProfile(GetProfile())->Classify(` so why aren't we just setting the correct url from the AutocompleteClassifier?

Reply:
`AutocompleteClassifierFactory` always uses normal profile.
So we did https://github.com/brave/brave-core/pull/5407/files#diff-ffcd3bd1d7f1c7ee80c4637a3f919142L10. This PR reverts it and instead set correct url by patching here.

Reply:
@bridiver PTAL again.